libs/domain/picking/src/components/user-profile/user-profile.component.spec.ts

```diff
@@ -0,0 +1,219 @@
+}
+
+class MockSyncSchedulerService implements Partial<SyncSchedulerService> {
+  hasPending = vi.fn().mockReturnValue(of(false));
+}
+
+class MockRouterService implements Partial<RouterService> {
+  route = vi.fn().mockReturnValue(of('/'));
+  navigate = vi.fn();
+}
+
+class MockAuthService implements Partial<AuthService> {
+  logout = vi.fn();
+}
+
+describe('UserProfileComponent', () => {
+  let element: UserProfileComponent;
+  let routerService: MockRouterService;
+  let syncSchedulerService: MockSyncSchedulerService;
+  let authService: MockAuthService;
+
+  const getReceiveDataButton = () => {
+    return element.renderRoot.querySelectorAll('button')[1];
+  };
+
+  beforeAll(async () => {
+    await useComponent(userProfileComponent);
+  });
+
+  beforeEach(async () => {
+    const testInjector = createInjector({
+      providers: [
+        {
+          provide: AppRef,
+          useClass: MockApp,
+        },
+        {
+          provide: RouterService,
+          useClass: MockRouterService,
+        },
+        {
+          provide: SyncSchedulerService,
+          useClass: MockSyncSchedulerService,
+        },
+        {
+          provide: AuthService,
+          useClass: MockAuthService,
+        },
+      ],
+    });
+
+    routerService = testInjector.inject(
+      RouterService
+    ) as unknown as MockRouterService;
+    syncSchedulerService = testInjector.inject(
+      SyncSchedulerService
+    ) as unknown as MockSyncSchedulerService;
+    authService = testInjector.inject(
+      AuthService
+    ) as unknown as MockAuthService;
+
+    element = await fixture(
+      html`<oryx-picking-user-profile></oryx-picking-user-profile>`
+    );
+  });
+
+  afterEach(() => {
+    destroyInjector();
+    vi.clearAllMocks();
+  });
+
+  it('passes the a11y audit', async () => {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  it('should be provided', () => {
+    expect(element).toBeInstanceOf(UserProfileComponent);
+  });
+
+  it('should render its components', () => {
+    expect(element.renderRoot.querySelector('.info-block')).not.toBeNull();
+    expect(
+      element.renderRoot.querySelector('.info-footer oryx-button')
+    ).not.toBeNull();
+  });
+
+  describe('when sync is pending', () => {
+    beforeEach(async () => {
+      syncSchedulerService.hasPending.mockReturnValue(of(true));
+
+      element = await fixture(
+        '<oryx-picking-user-profile></oryx-picking-user-profile>'
+      );
+    });
+
+    it('should render oryx notification', () => {
+      const notification =
+        element.renderRoot.querySelector('oryx-notification')?.textContent;
+      expect(notification).toContain('You can');
+      expect(notification).toContain(
+        't log out because of a pending synchronization'
+      );
+    });
+
+    it('should disable log out button', () => {
+      expect(
+        element.renderRoot.querySelector('button[disabled]')
```

Comment:
This checks that there is any generic button disabled and not exactly a logout button, please make a selector more specific for logout button.